wagtail/admin/views/generic/models.py

```diff
@@ -16,15 +17,65 @@ class IndexView(PermissionCheckedMixin, WagtailAdminTemplateMixin, BaseListView)
     index_url_name = None
     add_url_name = None
     edit_url_name = None
+    template_name = 'wagtailadmin/generic/index.html'
     context_object_name = None
     any_permission_required = ['add', 'change', 'delete']
+    page_kwarg = 'p'
+    default_ordering = None
+
+    def get(self, request, *args, **kwargs):
+        if not hasattr(self, 'columns'):
+            self.columns = self.get_columns()
+
+        return super().get(request, *args, **kwargs)
+
+    def get_columns(self):
```

Comment:
Would all index views be required to use this? Would it make sense to make this a mixin?

Reply:
I'd say it's as much a requirement as `get_form` is for an edit view... it's a core part of what the view is designed to do, and the expectation is that you will use it, but you have a decent amount of freedom to subclass and override it if you have non-standard requirements. If you don't want the provided table functionality, then I think you'd just swap out the template with one that doesn't use the `table` variable.

I think it's more likely that you'd just roll your own view class based on `WagtailAdminTemplateMixin` in that case though - there's nothing in viewsets or similar that requires you use IndexView, right?